src/v/cloud_storage/partition_manifest.cc

```diff
@@ -263,6 +265,61 @@ std::optional<model::offset> partition_manifest::get_start_offset() const {
     return _start_offset;
 }
 
+std::optional<kafka::offset>
+partition_manifest::get_start_kafka_offset() const {
+    if (_start_offset != model::offset{}) {
+        auto iter = _segments.find(_start_offset);
+        if (iter != _segments.end()) {
+            auto delta = iter->second.delta_offset;
+            return _start_offset - delta;
+        } else {
+            throw std::runtime_error(fmt_with_ctx(
+              fmt::format,
+              "can't translate start offset {} because the manifest is empty",
+              _start_offset));
+        }
+    }
+    return std::nullopt;
+}
+
+partition_manifest::const_iterator
+partition_manifest::segment_containing(kafka::offset o) const {
+    vlog(cst_log.debug, "Metadata lookup using kafka offset {}", o);
+    // Kafka offset is always <= log offset.
+    // To find a segment by its kafka offset we can simply query
+    // manifest by kafka offset and then traverse forward until we
```

Comment:
nit: should this comment say `query manifest by log/rp offset`?

Reply:
fixed